core/src/main/java/org/opensearch/sql/expression/function/BuiltinFunctionRepository.java

```diff
@@ -64,25 +67,45 @@ public class BuiltinFunctionRepository {
    *
    * @return singleton instance
    */
-  public static synchronized BuiltinFunctionRepository getInstance() {
-    if (instance == null) {
-      instance = new BuiltinFunctionRepository(new HashMap<>());
+  public static synchronized BuiltinFunctionRepository getInstance(
+      DataSourceService dataSourceService) {
+    Set<DataSourceMetadata> dataSourceMetadataSet = dataSourceService.getDataSourceMetadata(true);
+    Set<Integer> dataSourceServiceHashSet =
+        dataSourceMetadataSet.stream()
+            .map(metadata -> metadata.hashCode())
+            .collect(Collectors.toSet());
+
+    // Creates new Repository for every dataSourceService
+    if (!dataSourceServiceHashSet.stream().anyMatch(hash -> instance.containsKey(hash))) {
+      BuiltinFunctionRepository repository = new BuiltinFunctionRepository(new HashMap<>());
 
       // Register all built-in functions
-      ArithmeticFunction.register(instance);
-      BinaryPredicateOperator.register(instance);
-      MathematicalFunction.register(instance);
-      UnaryPredicateOperator.register(instance);
-      AggregatorFunction.register(instance);
-      DateTimeFunction.register(instance);
-      IntervalClause.register(instance);
-      WindowFunctions.register(instance);
-      TextFunction.register(instance);
-      TypeCastOperator.register(instance);
-      SystemFunctions.register(instance);
-      OpenSearchFunctions.register(instance);
```

Comment:
What happened to line 83?

Reply:
It's gone.  
OpenSearch functions are now registered in the OpenSearch module as part of the OpenSearch data source engine.